src/shr_abort_mod.F90

```diff
@@ -55,20 +54,17 @@ subroutine shr_abort_abort(string,rc)
     else
        local_string = "Unknown error submitted to shr_abort_abort."
     end if
+    if(present(rc)) then
+       write(local_string, *) trim(local_string), ' rc=',rc
+    endif
 
     call print_error_to_logs("ERROR", local_string)
 
+    call ESMF_LogWrite(local_string, ESMF_LOGMSG_ERROR, line=line, file=file)
```

Comment:
@jedwards4b question. What will happen if line and/or file aren't present in the call and they are sent down this way to ESMF_LogWrite? It seems like it might be compiler dependent on how it handles passing down a non-present argument.

But, I'm interested if you did any testing around that. Thanks.

Reply:
According to the standard: If you pass an omitted dummy argument as the actual argument to a procedure, the corresponding dummy argument is considered to be omitted as well. 

We do this in many places in cesm, as far as I know all of our compilers behave correctly.

Reply:
Awesome, good to know!